### DISCUSSION TITLE
Match simultaneous scopes when theming

### DISCUSSION THREAD

**Discussion by justinlovinger** [helix-editor/helix#10515] [General]
For example

```
"markup.quote" = "gray"
"markup.raw" = "gray"
"markup.quote markup.raw" = "white"
```

would style raw as gray, unless it was in a quote block, in which case it would be white.

**Comment by chtenb** [comment]
This is also how many other text editors do it, like VSCode, sublime, TextMate etc. Seem like a logical request to me, as it enables a more granular and modular way to describe highlighting rules. One cool special use case of this is JSON syntax highlighting which changes color depending on the nestedness: https://github.com/chtenb/neo-ansi/blob/main/templates/neo-ansi.json.yaml

![image](https://github.com/helix-editor/helix/assets/1030961/955f8a90-80cf-4e12-9e7b-b6dbcfafb089)
